### ISSUE TITLE
withAccelerate causes unhandled runtime error in nextjs server actions

### ISSUE THREAD

**Issue by edi** [prisma/prisma#25040]
### Bug description

Hi. Made a simple CRUD panel to test `withAccelerate` extension. However, even if I am not using it within client components, as the system obviously works just fine without it, I get the following error:

![CleanShot 2024-08-17 at 17 37 12](https://github.com/user-attachments/assets/d5c968ad-faf8-4100-92cf-b28acbe49765)

Removing the `withAccelerate` line of code works just fine, I'm confused as to what the next step is. The implementation is correct, however I assume that this error is triggered due to the client component importing a few server actions that require prisma.

### How to reproduce

1. Implement prisma accelerate
2. code a prisma query in a server action
3. import a server action into a client component

### Expected behavior

The above error should not trigger since this is not executed in the client.

`Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in unknown).`

Executing the same exact code without `Accelerate` works just fine.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
}

model Organisation {
  id        Int       @id @default(autoincrement())
  name      String
  users     User[]
  stations  Station[]
  updatedAt DateTime  @updatedAt
  createdAt DateTime  @default(now())
}

model Station {
  id        Int          @id @default(autoincrement())
  orgId     Int
  name      String
  updatedAt DateTime     @updatedAt
  createdAt DateTime     @default(now())
  org       Organisation @relation(fields: [orgId], references: [id])
}

model User {
  id        Int          @id @default(autoincrement())
  email     String       @unique
  name      String
  password  String
  orgId     Int
  updatedAt DateTime     @updatedAt
  createdAt DateTime     @default(now())
  org       Organisation @relation(fields: [orgId], references: [id])
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v20.14.0

### Prisma Version

```
prisma                  : 5.18.0
@prisma/client          : 5.18.0
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.14.0
Query Engine (Node-API) : libquery-engine 4c784e32044a8a016d99474bd02a3b6123742169 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 4c784e32044a8a016d99474bd02a3b6123742169 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.18.0-25.4c784e32044a8a016d99474bd02a3b6123742169
Default Engines Hash    : 4c784e32044a8a016d99474bd02a3b6123742169
Studio                  : 0.502.0
```

**Comment by Weakky** [comment]
Hey @edi, would you be able to provide us with a reproduction repository? That would be extremely helpful. Thank you

**Comment by edi** [comment]
Hi, I sure can, will take some time tomorrow to set something up but it's pretty straight forward. A simple server action that includes prisma, when imported into a client component, triggers the above error if using `withAccelerate` extension while working just fine if you don't use the extension.

I'll update you guys here tomorrow nonetheless, regards.

**Comment by SevInf** [comment]
Hi @edi.
Were you able to create the repository? 
Thank you!

**Comment by edi** [comment]
Hi! The issue persists and can easily ne replicated, I just am with my
second born kid now, so not much spare time. I’ll try sort it out tomorrow
🙏

**Comment by SevInf** [comment]
@edi congrats on the kid!

**Comment by jkomyno** [comment]
Hey @edi, do you have any update on this? Without a reproduction, we can't move forward. Thanks!

Ps, I hope the kid is well :)

**Comment by jkomyno** [comment]
Btw, this is potentially a duplicate of https://github.com/prisma/prisma/issues/24472

**Comment by edi** [comment]
Hi, sorry for the delay we've had our newborn. Yes, the #24472 issue has a more "vague" description, but it is exactly the same issue. Basically, if you have a server action that triggers any db query, as soon as you call it from a client component (server component works fine), the error shows up.

Let me create a reproductible example now.

**Comment by edi** [comment]
Hi. I've spent the last hour making a minimal example, with the same outcome. However, while working on it, I figured out the issue. 

_TLDR: Basically, it's not a bug, just inconsistent behaviour._

Let's take the following "utility" file as example:

```javascript
// utils.js
export const prisma = new PrismaClient()
export const accelerate = new PrismaClient().$extends(withAccelerate())
```

Let's consider a `client component` that (mistakenly) imports the `prisma client`, this works just fine and does NOT trigger any error until you actually run any prisma method, as expected.

```javascript
// client component (works fine)
import {prisma} from 'utils'

// client code
```

Now, let's consider the following client component, which also "mistakenly" imports `accelerate`.

```javascript
// client component (throws error on mount)
import {accelerate} from 'utils'

// client code
```

This is what causes the above error, instantly. As soon as the file that exports the prisma client is imported into any client component, even if you DON'T run any prisma-related code, it will throw an error.

This behaviour is what's different between the default prisma client and the one using the `accelerate` extension.

In my case, the confusion was created by the fact that I wasn't actually importing any prisma client in any client component, but I was importing something else from the same file that exported the prisma client. Specifically, I had a "cn" method (tailwind helper), exported from the same `utils.js` file where the prisma client is exported.

As soon as I ran `import {cn} from '@/utils'` the app crashed. I didn't notice this  because the `cn import` was already in the client components, and the `accelerate` extension was enabled later.

Let me know if the explanation above is clear enough or you need me to do reproducible example.

Regards

**Comment by nurul3101** [comment]
Hey @edi 👋 

I have prepared a minimal reproducible example: https://github.com/nurul3101/prisma-issue-25040

In this example, the `PrismaClient` (with and without Accelerate) is imported into a client component without triggering any errors and the application functions normally.

However, when the imported PrismaClient instance is actively used (for instance, by logging it), the error "PrismaClient  is unable to run in the browser" error is produced, as expected.

Can you check it out?

**Comment by nurul3101** [comment]
Hey @edi 

We tried to reproduce the issue but were unable to replicate the behavior described. Here's the link to reproduction: https://github.com/nurul3101/prisma-issue-25040

If you're still experiencing this problem, please feel free to open a new issue with additional details, and we'll be happy to investigate further.
